docs/proposals/job_controller.md

```diff
@@ -0,0 +1,226 @@
+* the job name ("jobName")
+* A tag ("jobTag", either a generated UID, or simply the schedule date of this job execution) allowing to group all pods started started for a given job execution.
+
+Along with labels, specific environment variables are also dynamically added in created pods definition: the number of pods started for this job execution and for each one its index number in the started pods list.
+
+Job has the responsibility to advertise job run completion status (success or failure) through events, to perform the necessary clean-up in pods registry, to provide a detailed status of active job runs and the completion status of previous completed runs up to a given amount of history).
+Collecting the standard output/error of pod's containers is not covered by this design (a common solution for containers started by any controller is needed).
+
+**API types definition:**
+
+The following API objects can be introduced:
+
+```
+type JobScheduleSpec struct {
+
+    // TimeSpec contains the schedule in ISO8601 format.
+    TimeSpec string `json:"timeSpec"`
+
+    // SkipOutdated specifies that if AllowConcurrent is false, only the latest pending job
+    // that is waiting for the currently executing one to complete must be started (default: true)
+    SkipOutdated bool `json:"skipOutdated"`
+    
+    // BlockOnFailure suspends scheduling of next job runs after a failed one
+    BlockOnFailure bool `json:"blockOnFailure"`
+}
+
+type JobSpec struct {
+
+    // JobScheduleSpec contains scheduling related infos
+    JobScheduleSpec ScheduleSpec `json:"scheduleSpec"`
+
+    // AllowConcurrent specified whether concurrent jobs may be started
+    // (covers the case where new job schedule time is reached, while
+    // other previously started jobs are still running)
+    AllowConcurrent bool `json:"allowConcurrent"`
+
+    // ExecTimeout specifies the max amount of time a job is allowed to run (in seconds).
+    ExecTimeout int `json:"execTimeout"`
+
+    // MaxRestart specifies the max number of restarts for a failing scheduled pod.
+    MaxRestart int `json:"maxRestart"`
+    
+    // PodCount specifies the numbe of pods to start for each job run
+    PodCount int `json:"podCount"`
+    
+    // Pod selector for that controller
+    Selector map[string]string `json:"selector"`
+
+    // Reference to stand alone PodTemplate
+    TemplateRef *ObjectReference `json:"templateRef,omitempty"`
+
+    // Embedded pod template
+    Template *PodTemplateSpec `json:"template,omitempty"`
+}
+
+type JobExecStatus string 
+const (
+    JobExecPending JobExecStatus   = "pending"
+    JobExecRunning JobExecStatus   = "running"
+    JobExecCompleted JobExecStatus = "completed"
+)
+
+type JobExecution struct {
+    // CreatedAt gives job creation time
+    CreatedAt util.Time
+
+    // StartedAt gives job starting time
+    StartedAt util.Time
+
+    // CompletedAt
+    CompletedAt util.Time
+
+    // Success gives job completion result
+    Success bool
```

Comment:
The next one, namely `Status` does that already, I don't see any point in repeating this as `bool` here.